recursive.c

```diff
@@ -0,0 +1,94 @@
+#include "recursive.h"
+
+#include <unistd.h>
+
+static bool
+bruteforce_rec_internal(struct task_t *task,
+                        struct config_t *config,
+                        void *context,
+                        password_handler_t handler,
+                        int pos)
+{
+    if (pos == task->to)
+    {
+        return handler(context, task);
+    }
+    else
+    {
+        for (int i = 0; config->alphabet[i] != '\0'; ++i)
+        {
+            task->password[pos] = config->alphabet[i];
+            if (bruteforce_rec_internal(task, config, context, handler, pos + 1))
+                return true;
+        }
+    }
+    return false;
+} 
+
+bool
+bruteforce_rec(struct task_t *task,
```

Comment:
Why context is void* here? It could be a typed pointer.
There is no need to pass handler as a parameter, because we use here only one handler and it could be called statically.
It make sense to have the following parameters: config, task, context. There is no need to store task in the context in this case.

Reply:
If I tweak process_task as suggested in comment above, then bruteforce_rec will be called with mt_password_handler from procedure `multithreaded`.

Reply:
Sorry, commented wrong function. I meant the same comment for the bruteforce_rec_cooperative

Reply:
Field task from rec_state_t is accessed in gn_worker. This could be refactored, but I find this architecture quite convenient.